SerialPrograms/Source/PokemonSV/Inference/Pokedex/PokemonSV_PokedexMenuDetector.cpp

```diff
@@ -0,0 +1,198 @@
+/*  Pokedex Menu Detector
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "Common/Cpp/AbstractLogger.h"
+#include "Kernels/Waterfill/Kernels_Waterfill_Session.h"
+#include "CommonFramework/Globals.h"
+#include "CommonFramework/ImageTools/BinaryImage_FilterRgb32.h"
+#include "CommonFramework/ImageMatch/ExactImageMatcher.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "PokemonSV_PokedexMenuDetector.h"
+
+//#include <iostream>
+//using std::cout;
+//using std::endl;
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+using namespace Kernels::Waterfill;
+
+
+const ImageMatch::ExactImageMatcher& RECENTLY_BATTLED_CROPPED(){

```

Comment:
Looks like a lot of copy-pasted stuff from the platform bot that isn't needed here?

Reply:
Cleaned up so only detection related code is left